lighthouse-core/report/html/renderer/dom.js

```diff
@@ -213,27 +213,42 @@ class DOM {
   /**
    * Guaranteed context.querySelector. Always returns an element or throws if
    * nothing matches query.
+   * @template {string} T
    * @param {string} query
    * @param {ParentNode} context
-   * @return {!HTMLElement}
+   * @param {T=} tagName
```

Comment:
can you add a description here.. like "Functionally optional, but providing it avoids needing a typecast"